libs/MeshKernel/include/MeshKernel/RangeCheck.hpp

```diff
@@ -0,0 +1,501 @@
+        ///                      (It is recommended to use a concise string)
+        template <RangeCheckableType T>
+        inline static void CheckRange(T const& value,
+                                      T const& bound,
+                                      std::function<bool(T const&, T const&)> predicate,
+                                      Comparison const comparison,
+                                      std::string_view const variable_name)
+        {
+            if (!predicate(value, bound))
+            {
+                throw RangeError(
+                    fmt_ns::format("{{}} = {{}} is invalid. Valid range: {}.", ValidRangeFormat.at(comparison)),
+                    variable_name,
+                    value,
+                    variable_name,
+                    bound);
+            }
+        }
+
+        /// @brief Checks the validity of a value given interval bounds, supports the predicates ==, !=, >, >=, <, and <=
+        /// @tparam T The type of the value to check (RangeCheckableType)
+        /// @param value The value to check
+        /// @param interval_lower_bound The lower bound of the interval
+        /// @param interval_upper_bound The upper bound of the interval
+        /// @param predicate The comparison predicate (one of <= ... <=, < ... <, < ... <=, and <= ... <)
+        /// @param comparison The key of the performed comparison
+        /// @param variable_name The name or description of the variable whose value is checked
+        ///                      (It is recommended to use a concise string)
+        template <RangeCheckableType T>
+        inline static void CheckRange(T const& value,
+                                      T const& interval_lower_bound,
+                                      T const& interval_upper_bound,
+                                      std::function<bool(T const&, T const&, T const&)> predicate,
+                                      Comparison const comparison,
+                                      std::string_view const variable_name)
+        {
+            if (!predicate(value, interval_lower_bound, interval_upper_bound))
+            {
+                throw RangeError(
+                    fmt_ns::format("{{}} = {{}} is invalid. Valid range: {}.", ValidRangeFormat.at(comparison)),
+                    variable_name,
+                    value,
+                    interval_lower_bound,
+                    variable_name,
+                    interval_upper_bound);
+            }
+        }
+
+        /// @brief Checks the validity of a value given a vector of values, supports the predicates one of and none of
+        /// @tparam T The type of the value to check (RangeCheckableType)
+        /// @param value The value to check
+        /// @param values The vector of values
+        /// @param predicate The comparison predicate (one of <= ... <=, < ... <, < ... <=, and <= ... <)
+        /// @param comparison The key of the performed comparison
```

Comment:
Incorrect predicate comment

Reply:
One question about the IsOneOf and IsNoneOf.
It probably will not occur: what if floating point numbers are being checked and one of the values, either the value to check or the set of number it will be checked against is not a literal but the result of some calculation. Can you be sure that you will get strict equality?

Reply:
Does the unit test answer your question?